sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultPlanGenerator.java

```diff
@@ -56,31 +55,62 @@ protected Phase from(RawPhase rawPhase, String phaseName, Collection<PodSpec> po
                 List<String> taskNames = podSpec.getTasks().stream()
                         .map(taskSpec -> taskSpec.getName())
                         .collect(Collectors.toList());
-
-                // If the tasks to be launched have been explicitly indicated in the plan
-                // override the taskNames.
-                if (!CollectionUtils.isEmpty(rawPhase.getTasks())) {
-                    taskNames = rawPhase.getTasks();
-                }
-
                 steps.add(from(new DefaultPodInstance(podSpec, i), taskNames));
             }
         } else {
-            for (Map<Integer, List<String>> rawStepMap : rawPhase.getSteps()) {
-                if (rawStepMap.size() != 1) {
-                    throw new IllegalStateException(String.format(
-                            "Malformed step in phase '%s': Map should contain a single entry, but has %d: %s",
-                            phaseName, rawStepMap.size(), rawStepMap));
+            // Guarantee each map has exactly one element
+            List<WriteOnceLinkedHashMap<String, List<List<String>>>> rawSteps = rawPhase.getSteps();
+            validateSingletonStepMaps(phaseName, rawSteps);
+
+            // Convert from map to list
+            Map<String, List<List<String>>> validatedSteps =
+                    rawSteps.stream()
+                            .map(stepMap -> stepMap.entrySet().stream().findFirst().get())
+                            .collect(Collectors.toMap(
+                                    stringListEntry -> stringListEntry.getKey(),
+                                    stringListEntry -> stringListEntry.getValue()));
+
+            for (int i = 0; i < podSpec.getCount(); i++) {
+                List<List<String>> taskLists = validatedSteps.get(String.valueOf(i));
+                if (taskLists == null) {
+                    taskLists = validatedSteps.get("default");
+
+                    if (taskLists == null) {
+                        // Fall back to default behavior
+                        List<String> taskNames = podSpec.getTasks().stream()
+                                .map(taskSpec -> taskSpec.getName())
+                                .collect(Collectors.toList());
+                        steps.add(from(new DefaultPodInstance(podSpec, i), taskNames));
+                    } else {
+                        // Use wildcard defined behavior (e.g. * : [[foo, bar], [bazz]])
```

Comment:
My mistake.  Tried to use `*` instead of `default` initially.  I'll fix the comment.